### PR TITLE
Interfaces Update

### PR DESCRIPTION
**Pull request by Anyhowclick**
As per drafted [here](https://gist.github.com/manhlx3006/0be1b92a6328a4ccc39b951f655d21b8)

### CODE REVIEW

File: contracts/interfaces/IKyberGovernance.sol [hunk 1]
```diff
@@ -2,85 +2,118 @@
...
+    uint256 endTimestamp;
+    uint256 executionTimestamp;
+    string link;
     bool executed;
     bool canceled;
-    address strategy;
-    bytes32 ipfsHash;
   }
 
+  function handleVotingPowerChanged(address staker, uint256 newVotingPower) external;
```
Comment (manhlx3006) [inline contracts/interfaces/IKyberGovernance.sol:65]: please add `uint256[] calldata proposalIds` as a function param

File: contracts/interfaces/IVotingPowerStrategy.sol [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    /// pass creator in case we want to validate if creator has enough quorum to create proposal
+    function validateProposalCreation(
+      address creator,
+      uint256 startTimestamp,
+      uint256 endTimestamp
+    ) external view returns (bool);
+
+    /// we could have a function to validate if sender has enough quorum to cancel
+    /// but not necessary at this phase
```
Comment (manhlx3006) [inline contracts/interfaces/IVotingPowerStrategy.sol:40]: Please add getMaxVotingPower() function

File: contracts/interfaces/IVotingPowerStrategy.sol [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.5;
+pragma abicoder v2;
+
+interface IVotingPowerStrategy {
+    function handleProposalCreation(
+      uint256 proposalID,
```
Comment (manhlx3006) [inline contracts/interfaces/IVotingPowerStrategy.sol:7]: Please change to proposalId for consistency

File: contracts/interfaces/IKyberGovernance.sol [hunk 4]
```diff
@@ -2,85 +2,118 @@
...
+    uint256 startTimestamp;
+    uint256 endTimestamp;
+    uint256 executionTimestamp;
+    string link;
     bool executed;
     bool canceled;
-    address strategy;
-    bytes32 ipfsHash;
     mapping(address => Vote) votes;
   }
```
Comment (ducquangkstn) [inline contracts/interfaces/IKyberGovernance.sol:40]: To simplify, I prefer
```solidity
struct Proposal {
  ProposalWithoutVotes dataWithoutVote;
  mapping(address => Vote) votes;
}
```

Comment (ducquangkstn) [reply contracts/interfaces/IKyberGovernance.sol:40]: I also want to separate structs for binary proposal and general proposal
```solidity
struct BinaryProposalData {
  address[] targets;
  uint256[] values;
  string[] signatures;
  bytes[] calldatas;
  bool[] withDelegatecalls;
}

struct GeneralProposalData {
  ...
}

struct campaign {
  BinaryProposalData binaryData;
  GeneralProposalData generalData;
}
```

File: contracts/interfaces/IKyberGovernance.sol [hunk 5]
```diff
@@ -162,32 +220,9 @@ interface IAaveGovernanceV2 {
...
-    bytes32 s
-  ) external;
-
-  /**
-   * @dev Set new GovernanceStrategy
-   * Note: owner should be a timelocked executor, so needs to make a proposal
-   * @param governanceStrategy new Address of the GovernanceStrategy contract
-   **/
-  function setGovernanceStrategy(address governanceStrategy) external;
+  function submitVote(uint256 proposalId, uint256 optionBitMask) external;
```
Comment (ducquangkstn) [inline contracts/interfaces/IKyberGovernance.sol:230]: Pls add TODO `submitVoteBySignature`

File: contracts/interfaces/IKyberGovernance.sol [hunk 6]
```diff
@@ -2,85 +2,122 @@
...
 pragma abicoder v2;
 
-import {IExecutorWithTimelock} from './IExecutorWithTimelock.sol';
+import {IExecutorWithTimelock} from "./IExecutorWithTimelock.sol";
+import {IVotingPowerStrategy} from "./IVotingPowerStrategy.sol";
 
-interface IAaveGovernanceV2 {
+
+interface IKyberGovernance {
   enum ProposalState {Pending, Canceled, Active, Failed, Succeeded, Queued, Expired, Executed}
```
Comment (ducquangkstn) [inline contracts/interfaces/IKyberGovernance.sol:10]: Succeeded and Failed should be concat into 1

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**